### PR TITLE
Monetization (MON-367): MonetizationModule - added MonetizationService to AdEngine

### PR DESCRIPTION
**Pull request by saipetch**
Monetization (MON-367): add MonetizationService to AdEngine and update unit tests
(replace #7045)

### CODE REVIEW

File: extensions/wikia/AdEngine/AdEngine2ContextService.class.php [hunk 1]
```diff
@@ -24,6 +24,11 @@ public function getContext( Title $title, $skinName ) {
 				$sevenOneMediaCombinedUrl = ResourceLoader::makeCustomURL( $wg->Out, ['wikia.ext.adengine.sevenonemedia'], 'scripts' );
 			}
 
+			$monetizationServiceAds = null;
+			if ( !empty( $wg->AdDriverUseMonetizationService ) && !empty( $wg->EnableMonetizationModuleExt ) ) {
+				$monetizationServiceAds = F::app()->sendRequest( 'MonetizationModule', 'index' )->getData()['data'];
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/AdEngine2ContextService.class.php]: I'm going to nitpick a little bit. I hope you'll forgive me :japanese_ogre: 

What happens when there is error or empty response from Monetization service?

File: extensions/wikia/AdEngine/js/config/desktop.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ define('ext.wikia.adEngine.config.desktop', [
 	'ext.wikia.adEngine.provider.evolve',
 	'ext.wikia.adEngine.provider.liftium',
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/config/desktop.js]: Note to Ad Engineering: liftium is placed wrongly. Next time we touch `config/desktop.js` we could move it down so we keep alphabetical order.

File: extensions/wikia/AdEngine/js/provider/monetizationService.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	function fillInSlot(slot, success) {
+		log(['fillInSlot', slot], 'info', logGroup);
+
+		var slotName = slotMap[slot],
+			context = adContext.getContext();
+
+		if (context.providers.monetizationServiceAds && context.providers.monetizationServiceAds[slotName]) {
+			log(['fillInSlot', slot, 'injectScript'], 'info', logGroup);
+
+			monetizationService.loadAssets();
```
Comment (nandy-andy) [inline extensions/wikia/AdEngine/js/provider/monetizationService.js]: If I understand correctly the assets are cached so the request for them is being sent only once: for the first slot. What do you think about placing a comment about it above?

File: skins/oasis/modules/templates/Body_Index.php [hunk 4]
```diff
@@ -139,9 +139,9 @@
 				?>
 
 				<?php
-					if ( !empty( $wg->EnableMonetizationModuleExt ) && !empty( $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY] ) ) {
-						echo $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY];
-					}
+				if ( !empty( $wg->EnableMonetizationModuleExt ) && !empty( $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY] ) ) {
+					echo $monetizationModules[MonetizationModuleHelper::SLOT_TYPE_BELOW_CATEGORY];
```
Comment (nandy-andy) [inline skins/oasis/modules/templates/Body_Index.php]: Why don't we have here the same logic as below for `MON_ABOVE_FOOTER`?

File: extensions/wikia/AdEngine/AdEngine2ContextService.class.php (hunk 1 above)
Comment (jimmordino) [inline extensions/wikia/AdEngine/AdEngine2ContextService.class.php]: [Errors](http://graph-s3/grafana/#/dashboard/db/Monetization%20Service?panelId=5&fullscreen&from=now-24h&to=now)?  :smiling_imp:

**Comment by gabrys** [comment]
@saipetch I updated the pull request with a few tweaks, see the commit description

**Comment by gabrys** [comment]
@saipetch: is there any reason for this PR being against MonetizationModule, not dev?

And one more question: the loadAssets is asynchronous and will be loaded in parallel to the actual Amazon/AdSense creatives. Is there a need to synchronize those, so that the ads load after the CSS and JS are loaded and executed (or the other way round)?

File: skins/oasis/modules/templates/Body_Index.php (hunk 4 above)
Comment (saipetch) [inline skins/oasis/modules/templates/Body_Index.php]: Because below category slot uses different rule.

**Pull request closed without merging**